src/vm/database/sqlite.rs

```diff
@@ -296,23 +326,92 @@ impl <'a> ContractDatabase <'a> {
         }
     }
 
-    pub fn get_simmed_block_height(&self) -> Result<i128> {
-        let block_height: (i128) =
+    pub fn get_simmed_block_height(&self) -> Result<u64> {
+        let block_height: (i64) =
             self.query_row(
-                "SELECT simmed_block_height FROM simmed_block_table LIMIT 1",
+                "SELECT block_height FROM simmed_block_table ORDER BY block_height DESC LIMIT 1",
                 NO_PARAMS,
                 |row| row.get(0))
             .expect("Failed to fetch simulated block height");
 
-        Ok(block_height)
+        Ok(u64::try_from(block_height).unwrap())
```

Comment:
I'm not sure if this should panic.  It doesn't in `BurnDB` for example -- it just returns an `Overflow` error.  If it should panic, then you should use `panic!` or similar to print an explicit error message.